vaccine_feed_ingest/runners/tx/arcgis/normalize.py

```diff
@@ -0,0 +1,225 @@
+
+
+def _get_inventory(site: dict) -> Optional[List[schema.Vaccine]]:
+    inventory = []
+
+    for field, vaccine in VACCINES_FIELD.items():
+        try:
+            if site["attributes"][field] > 0 and vaccine not in inventory:
+                inventory.append(vaccine)
+        except KeyError:
+            pass
+
+    if len(inventory) > 0:
+        return inventory
+
+    return None
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    if site["attributes"]["PublicPhone"]:
+        sourcePhone = re.sub("[^0-9]", "", site["attributes"]["PublicPhone"])
+        if len(sourcePhone) == 11:
+            sourcePhone = sourcePhone[1:]
+        phone = f"({sourcePhone[0:3]}) {sourcePhone[3:6]}-{sourcePhone[6:]}"
+        contacts.append(schema.Contact(phone=phone))
+
+    if site["attributes"]["WEBSITE"]:
+        contacts.append(schema.Contact(website=site["attributes"]["WEBSITE"]))
+
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+
+def _get_published_at(site: dict) -> Optional[str]:
+    date = site["attributes"]["LAST_UPDATE_VAC"].strip()
+    time = site["attributes"]["LAST_UPDATE_TIME_VAC"].strip()
+    if date and time:
+        date_time_str = date + " " + time
+        date_time_obj = datetime.datetime.strptime(date_time_str, "%m/%d/%Y %H:%M:%S")
+        return date_time_obj.isoformat()
+
+    return None
+
+
+def _get_address(site: dict) -> Optional[schema.Address]:
+    if site["attributes"]["ADDRESS"] is None:
+        return None
+
+    address_field = site["attributes"]["ADDRESS"].replace(",", "").split(" ")
+    city_starts = 0
+    city_ends = 0
+
+    if site["attributes"]["CITY"] is None:
+        # Some sites put all address data in a single field.
+        # In this case, he data seems to uppercase all characters in the address.
+        # But the city only capitalizes the first letter
+        for index, field in enumerate(address_field):
+            try:
+                if len(field) > 1 and field[1].islower() and city_starts == 0:
+                    city_starts = index
+                if field == "TX":
+                    city_ends = index
+                if index == len(address_field) - 1 and city_starts == 0:
+                    city_starts = index - 1
+                    city_ends = index
+            except IndexError as ie:
+                logger.error("Unable to parse address: %s", ie)
+                return None
+
+        return schema.Address(
+            street1=" ".join(address_field[0:city_starts]),
+            street2=None,
+            city=" ".join(address_field[city_starts:city_ends]),
+            state="TX",
+            zip=address_field[-1],
+        )
+
+    else:
+        # Sometimes the zip can be None, even though the rest of the address has been entered
+        zip = (
+            site["attributes"]["ZIP"]
+            if site["attributes"]["ZIP"] is not None
+            else 00000
+        )
```

Comment:
@bryanculbertson - how would you like to handle this case? Should we make `zip` optional?

Reply:
@eliblock Yeah, let's make zip optional